src/UrlTwigExtension.php

```
@@ -0,0 +1,221 @@
+<?php
+
+namespace Massive\Component\Web;
+
+/**
+ * This Twig Extension manages url transformation.
+ */
+class UrlTwigExtension extends \Twig_Extension
+{
+    const DEFAULT_PROTOCOL = 'http';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return [
+            new \Twig_SimpleFilter('url_sanitize', [$this, 'sanitizeUrl']),
+            new \Twig_SimpleFilter('url_protocol', [$this, 'getProtocol']),
+            new \Twig_SimpleFilter('url_host', [$this, 'getHost']),
+            new \Twig_SimpleFilter('url_port', [$this, 'getPort']),
+            new \Twig_SimpleFilter('url_domain', [$this, 'getDomain']),
+            new \Twig_SimpleFilter('url_path', [$this, 'getPath']),
+            new \Twig_SimpleFilter('url_query', [$this, 'getQuery']),
+            new \Twig_SimpleFilter('url_fragment', [$this, 'getFragment']),
+        ];
+    }
+
+    /**
+     * Get all components of url.
+     *
+     * @param $url string
+     *
+     * @return array
+     *
+     * @throws \InvalidArgumentException
+     */
+    private static function parseUrl($url)
+    {
+        $parsedUrl = parse_url($url);
+
+        if (false === $parsedUrl) {
+            throw new \InvalidArgumentException(sprintf('URL (%s) is malformed!', $url));
```

twig extension should not throw exception as you can't catch exceptions in twig return empty string instead

---

doesn't twig ignore exceptions in production but show them in dev environment? with an exception developers would see that they made a mistake.

---

no also on production the exception is not ignored